src/nvim/api/vim.c

```diff
@@ -164,6 +165,39 @@ Object vim_eval(String str, Error *err)
   return rv;
 }
 
+/// Call the given function with the given arguments stored in an array.
+///
+/// @param functionName The function to call
+/// @param args the functions arguments packed in an Array
+/// @param[out] err Details of an error that may have occurred
+/// @return The result of the function call if it was successful
+Object vim_function(String functionName, Array args, Error *err)
+  FUNC_ATTR_DEFERRED
+{
+  Object rv = OBJECT_INIT;
+  typval_T r;
+  size_t i;
+  typval_T argv[args.size];
```

Comment:
I believe you need space for 1 extra argument, and it's type must be `VAR_UNKNOWN `.